### ISSUE TITLE
Question about the setting of "mode_confident"

### ISSUE THREAD

**Issue by Hugo-cell111** [xiaoyao3302/CCVC#4]
Hi! I wonder what the exact difference is among six settings of "mode_confident": ['normal', 'soft', 'vote', 'vote_threshold', 'vote_soft', 'no']. Looking forward to your reply and greatly appreciate your amazing work. Thanks!

**Comment by xiaoyao3302** [comment]
Hi, thanks for your attention! I forgot to delete the useless settings, sorry for the misleading. I have tried several cases for the second part (CPL) of our work, i.e., 'normal', 'soft', 'vote', 'vote_threshold', 'vote_soft', and 'no', please refer to utils.py for more details about the implementation of each case.

'no': we use all predictions from the other branch as the supervision to supervise the prediction logits of the selected branch.
'normal': we only use confident predictions as the supervision by comparing the prediction probability with the threshold.
'soft': we use confident predictions as the supervision with a co-efficient w1, and we also use unconfident predictions as the supervision with a co-efficient w2.
'vote': if two branches have the same predictions, we set the no-conflicting predictions as the supervision with a co-efficient w1, and we also use conflicting predictions as the supervision with a co-efficient w2.
'vote_threshold': we select conflicting but confident predictions and use the predictions as the supervision with a co-efficient w1, and we use other predictions as the supervision with a co-efficient w2.
'vote_soft': can be treated as the combination of 'vote_threshold' and 'soft', we select no-conflicting predictions as the supervision with a co-efficient w0 (which is set as 1 in the code), conflicting but confident predictions as the supervision with a co-efficient w1 and conflicting but unconfident predictions as the supervision with a co-efficient w2.

In one word, you just need to set mode_confident as 'vote_threshold'.

**Comment by Hugo-cell111** [comment]
Hi! As introduced in the paper, the final version is "vote_soft", and is the co-efficient w0=w2=1, w1=wc=2?

**Comment by xiaoyao3302** [comment]
Hi, please refer to train.sh, we used vote_threshold. 
Also refer to utils.py for details of vote_threshold and vote_soft.
We kept the codes as we did such experiments, maybe you can run different strategies and compare the performance. Based on our experiments, vote_threshold is better than vote_soft.

**Comment by Hugo-cell111** [comment]
ohohoh, my mistake. So sorry! Thank you!
